[do not review][branch-4.0] Pick #57724 and #57783

freemandealer opened the pull request:
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [ ] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

freemandealer committed:
[enhancement](filecache) use rocksdb to persist cache block meta (#57072)

freemandealer committed:
[enhancement](filecache) fine-grained cache space observation (#57783)

freemandealer commented:
run buildall

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/5b155531b68037986920787052a617d63e70acbc_5b155531b68037986920787052a617d63e70acbc_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/5b155531b68037986920787052a617d63e70acbc_5b155531b68037986920787052a617d63e70acbc_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 83.49% (1573/1884) |
| Line Coverage     | 67.65% (28046/41459) |
| Region Coverage   | 68.18% (13815/20264) |
| Branch Coverage   | 58.41% (7365/12610) |

freemandealer committed:
fix after rebase

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

yiguolei commented:
run buildall

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/8de5f792537267339153dcec8e8fc9084210d4f0_8de5f792537267339153dcec8e8fc9084210d4f0_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/8de5f792537267339153dcec8e8fc9084210d4f0_8de5f792537267339153dcec8e8fc9084210d4f0_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 83.49% (1573/1884) |
| Line Coverage     | 67.63% (28039/41459) |
| Region Coverage   | 68.17% (13813/20264) |
| Branch Coverage   | 58.43% (7368/12610) |

freemandealer committed:
[fix](filecache) fix out-of-range exception when external query

add check for external data (lakehouse data) when get_tablet_id
to avoid out-of-range exception

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer committed:
add tests

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer committed:
fix regression

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer committed:
[refactor](filecache) ttl management refactoring

freemandealer commented:
run buildall

freemandealer commented:
run buildall

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  100.00% (11/11)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/58025_54aeebb3607f380fb10e42fc0e130281719510be/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/58025_54aeebb3607f380fb10e42fc0e130281719510be/fe_report/index.html)

freemandealer committed:
[fix](filecache) fix clear directly rm meta dir

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer committed:
fix beut

Signed-off-by: freemandealer <freeman.zhang1992@gmail.com>

freemandealer committed:
rm invalid beut

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer commented:
run buildall

freemandealer committed:
fix beuts

Signed-off-by: freemandealer <freeman.zhang1992@gmail.com>

freemandealer committed:
fix observation

Signed-off-by: freemandealer <freeman.zhang1992@gmail.com>

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  100.00% (12/12)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/58025_2131442cb9267456e1d6e3a3695c91384875ba95/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/58025_2131442cb9267456e1d6e3a3695c91384875ba95/fe_report/index.html)

freemandealer committed:
Remove duplicate cache schema change hit ratio definition

Removed duplicate definition of file_cache_keep_schema_change_output_min_hit_ratio.

freemandealer committed:
[fix](filecache) fix fs iterator concurrency problem

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer committed:
fix reset_range not update shadow queue cause cache size large

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

freemandealer committed:
fix calc bug

Signed-off-by: zhengyu <zhangzhengyu@selectdb.com>

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  100.00% (12/12)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/58025_f47bc9e1ebaa66293ef37c45517ed3a5eee22504/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/58025_f47bc9e1ebaa66293ef37c45517ed3a5eee22504/fe_report/index.html)
